fili-core/src/main/java/com/yahoo/bard/webservice/data/config/dimension/DefaultDimensionConfig.java

```diff
@@ -0,0 +1,134 @@
+     * @param category  The Category is the external, end-user-facing category for the dimension.
+     * @param fields  The set of fields for this dimension.
+     * @param defaultDimensionFields  The default set of fields for this dimension to be shown in the response.
+     * @param keyValueStore  The key value store holding dimension row data.
+     * @param searchProvider  The search provider for field value lookups on this dimension.
+     */
+    public DefaultDimensionConfig(
+        @NotNull DimensionName apiName,
+        @NotNull String physicalName,
+        @NotNull String description,
+        @NotNull String longName,
+        @NotNull String category,
+        @NotNull LinkedHashSet<DimensionField> fields,
+        @NotNull LinkedHashSet<DimensionField> defaultDimensionFields,
+        @NotNull KeyValueStore keyValueStore,
+        @NotNull SearchProvider searchProvider
+    ) {
+      this.apiName = apiName;
+      this.physicalName = physicalName;
+      this.description = description;
+      this.longName = longName;
+      this.category = category;
+      this.fields = fields;
+      this.defaultDimensionFields = defaultDimensionFields;
+      this.keyValueStore = keyValueStore;
+      this.searchProvider = searchProvider;
+    }
+
+  /**
+   * Construct a DefaultDimensionConfig Instance from Dimension Name and only using default dimension fields.
+   *
+   * @param apiName  The API Name is the external, end-user-facing name for the dimension.
+   * @param physicalName  The internal, physical name for the dimension.
+   * @param description  A description of the dimension and its meaning.
+   * @param longName  The Long Name is the external, end-user-facing long  name for the dimension.
+   * @param category  The Category is the external, end-user-facing category for the dimension.
+   * @param defaultDimensionFields  The default set of fields for this dimension to be shown in the response.
+   * @param keyValueStore  The key value store holding dimension row data.
+   * @param searchProvider  The search provider for field value lookups on this dimension.
+   */
+  public DefaultDimensionConfig(
+      DimensionName apiName,
+      String physicalName,
+      String description,
+      String longName,
+      String category,
+      LinkedHashSet<DimensionField> defaultDimensionFields,
+      KeyValueStore keyValueStore,
+      SearchProvider searchProvider
+  ) {
+    this (
+        apiName,
+        physicalName,
+        description,
+        longName,
+        category,
+        defaultDimensionFields,
+        defaultDimensionFields,
+        keyValueStore,
+        searchProvider
+    );
+  }
+
+  public String getApiName() {
+    return apiName.asName();
+  }
+
+  public String getLongName() {
+    return longName;
+  }
+
+  public String getCategory() {
+    return category;
+  }
+
+  public String getPhysicalName() {
+    return physicalName;
+  }
+
+  public String getDescription() {
```

Comment:
All of these methods, if they override a method declared in the interface, should have an `@Override` annotation.